### PR TITLE
#855 Descriptive integrity check messages

### PR DESCRIPTION
**Pull request by Dzejkop**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs [hunk 1]
```diff
@@ -91,17 +91,20 @@ private FileIntegrity CheckFile(AppContentSummaryFile file)
...
+                return new FileIntegrity(file.Path, FileIntegrityStatus.InvalidVersion, message);
             }
 
             if (_isCheckingSize)
             {
                 long size = new FileInfo(localPath).Length;
                 if (size != file.Size)
                 {
-                    return new FileIntegrity(file.Path, FileIntegrityStatus.InvalidSize);
+                    string message = string.Format("Expected {0}, but is {1}", file.Size, size);
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs]: Please change the name of `size` variable to `actualSize`. I had to look what is assigned to that variable to know what is in it.

File: Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs [hunk 2]
```diff
@@ -91,17 +91,20 @@ private FileIntegrity CheckFile(AppContentSummaryFile file)
                 return new FileIntegrity(file.Path, FileIntegrityStatus.MissingMetaData);
             }
 
-            if (_localMetaData.GetEntryVersionId(file.Path) != _versionId)
+            int entryVersionId = _localMetaData.GetEntryVersionId(file.Path);
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs]: Please maybe change `entryVersionId` to `actualEntryVersionId`. It's easier to tell that way what the following if() is doing.

File: Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs [hunk 3]
```diff
@@ -110,7 +113,8 @@ private FileIntegrity CheckFile(AppContentSummaryFile file)
                 string hash = HashCalculator.ComputeFileHash(localPath);
                 if (hash != file.Hash)
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs]: Please change `hash` variable na to `actualFileHash`.

File: Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/FileIntegrity.cs [hunk 4]
```diff
@@ -2,14 +2,17 @@
...
+        public FileIntegrity(string fileName, FileIntegrityStatus status, string message = null)
         {
             FileName = fileName;
             Status = status;
+            Message = message;
         }
 
         public string FileName { get; private set; }
 
+        public readonly string Message;
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/FileIntegrity.cs]: I don't see any read usage of `Message` field within this PR. Is this a mistake?

Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/FileIntegrity.cs]: Yup, and it's corrected now

**Pull request merged**